src/test/java/de/hpi/bpt/argos/api/EntityEndpointTest.java

```diff
@@ -207,14 +208,50 @@ public void testGetEventsOfEntity_SwappedIndices_Success() {
 	}
 
 	@Test
-	public void testGetEventsOfEntity_InvalidEntityId_Success() {
+	public void testGetEventsOfEntity_ChildEvents_Success() {
+		Entity newEntity = ArgosTestUtil.createEntity(testEntityType, testEntity, true);
+		Event newEvent = ArgosTestUtil.createEvent(testEventType, newEntity, true);
+		List<Attribute> newEventAttributes = ArgosTestUtil.createEventAttributes(testEventType, newEvent, true);
+
 		RestRequest request = RestRequestFactoryImpl.getInstance()
-				.createGetRequest(ARGOS_REST_HOST, getEventsOfEntityUri(testEntity.getId() + 1, testEventType.getId(), 0, 999));
+				.createGetRequest(ARGOS_REST_HOST, getEventsOfEntityUri(testEntity.getId(), testEventType.getId(), 0, 999));
+
+		assertEquals(HttpStatusCodes.SUCCESS, request.getResponseCode());
+
+		// delete these again to not disturb other tests
+		PersistenceAdapterImpl.getInstance().deleteArtifacts(newEntity, newEvent);
+		PersistenceAdapterImpl.getInstance().deleteArtifacts(newEventAttributes.toArray(new Attribute[newEventAttributes.size()]));
+
+		JsonArray events = jsonParser.parse(request.getResponse()).getAsJsonArray();
+		assertEquals(2, events.size());
+
+		JsonObject event = events.get(0).getAsJsonObject();
+		JsonArray eventAttributes = event.get("Attributes").getAsJsonArray();
+		assertEquals(testEventAttributes.size(), eventAttributes.size());
+
+		event = events.get(1).getAsJsonObject();
+		eventAttributes = event.get("Attributes").getAsJsonArray();
+		assertEquals(newEventAttributes.size(), eventAttributes.size());
```

Comment:
How do you make sure the events are different? Those could as well be the attributes of the original testEventType.